dll/win32/shell32/utils.cpp

```diff
@@ -9,6 +9,90 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(shell);
 
+static PCSTR StrEndNA(_In_ LPCSTR psz, _In_ INT_PTR cch)
+{
+    PCSTR pch, pchEnd = &psz[cch];
+    for (pch = psz; *pch && pch < pchEnd; pch = CharNextA(pch))
+        ;
+    if (pchEnd < pch)
+        pch -= 2;
```

Comment:
Why "minus 2" ?

Reply:
The original authors` mistake?

Reply:
Who're the original authors? They should be you, not "somebody else"....
If you're the code author you should know how to explain to me what this minus 2 is for.

Reply:
The condition `pchEnd < pch` at here means there are a double-byte character.
The width of the double-byte character is 2.

Reply:
Comments added.